--- a/src/addrwatch_stdout.c	2020-02-03 00:22:18.000000000 +0100
+++ b/src/addrwatch_stdout.c	2021-06-24 20:27:47.682214123 +0200
@@ -16,7 +16,7 @@
 		ip4_ntoa(e->ip_address, ip_str);
 	}
 
-	printf("%lu %s %u %s %s %s\n", e->timestamp, e->interface, e->vlan_tag,
+	printf("%llu %s %u %s %s %s\n", e->timestamp, e->interface, e->vlan_tag,
 		mac_str, ip_str, pkt_origin_str[e->origin]);
 }
 
--- a/src/addrwatch_syslog.c	2020-02-03 00:22:18.000000000 +0100
+++ b/src/addrwatch_syslog.c	2021-06-24 20:27:55.486631139 +0200
@@ -18,7 +18,7 @@
 		ip4_ntoa(e->ip_address, ip_str);
 	}
 
-	syslog(LOG_INFO, "%lu %s %u %s %s %s", e->timestamp, e->interface,
+	syslog(LOG_INFO, "%llu %s %u %s %s %s", e->timestamp, e->interface,
 		e->vlan_tag, mac_str, ip_str, pkt_origin_str[e->origin]);
 }
 
